hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/Route.java

```diff
@@ -310,5 +358,39 @@ public void loadValidator() {
 			System.exit(0);
 		}
     	
-    }
+    }    
+
+	/**
+	 * Creates MQ connection and sets it on a JMS Component which is added to the camel context.
+	 */
+	protected void initMQ() {
+		final String methodName = LoggingUtil.getMethodName();
+		JmsComponent jmsComponent = new JmsComponent();
+    	MQQueueConnectionFactory mqQueueConnectionFactory = mqQueueConnectionFactory();
+		jmsComponent.setConnectionFactory(mqQueueConnectionFactory);
+        getContext().addComponent("jmsComponent", jmsComponent);
+		logger.info("{} - Added JMS Component to context with connection factory. {}",methodName);			
+	}
+    
+    /**
+     * Creates a MQ connection factory and sets its connection properties.
+     * 
+     * @return a {@link MQQueueConnectionFactory} factory
+     */
+    public MQQueueConnectionFactory mqQueueConnectionFactory()  {
+		final String methodName = LoggingUtil.getMethodName();
+        MQQueueConnectionFactory mqQueueConnectionFactory = new MQQueueConnectionFactory();
+      
+        mqQueueConnectionFactory.setHostName(System.getenv("MQ_HOST"));
+        try {
+        	mqQueueConnectionFactory.setTransportType(WMQConstants.WMQ_CM_CLIENT);         
+        	mqQueueConnectionFactory.setChannel(System.getenv("MQ_CHANNEL"));
+        	mqQueueConnectionFactory.setPort(Integer.valueOf(System.getenv("MQ_PORT")));
+        	mqQueueConnectionFactory.setQueueManager(System.getenv("MQ_QUEUEMANAGER"));
+    		logger.debug("{} - MQ connection factory has been created.",methodName);			
+        } catch (JMSException jmse) {
+        	 logger.error("{} - MQ connection factory initialization failed with the error : {}", methodName, jmse.getMessage());       	
```

Comment:
There appears to be an extra space after the tab indent.
Also, do we want to do anything else if a JMSException is thrown? Should HNS-ESB even start up?